src/control/lib/support/log.go

```diff
@@ -0,0 +1,640 @@
+//
+// (C) Copyright 2022 Intel Corporation.
+//
+// SPDX-License-Identifier: BSD-2-Clause-Patent
+//
+
+package support
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/daos-stack/daos/src/control/common"
+	"github.com/daos-stack/daos/src/control/lib/hardware"
+	"github.com/daos-stack/daos/src/control/lib/hardware/hwprov"
+	"github.com/daos-stack/daos/src/control/logging"
+	"github.com/daos-stack/daos/src/control/server/config"
+)
+
+// Folder names to copy logs and configs
+const (
+	dmgSystemLogFolder = "DmgSystemLog"     // Copy the dmg command output for DAOS system
+	daosNodeLogFolder  = "DaosNodeLog"      // Copy the dmg command output specific to the storage.
+	daosAgentCmdInfo   = "DaosAgentCmdInfo" // Copy the daos_agent command output specific to the node.
+	systemInfo         = "SysInfo"          // Copy the system related information
+	serverLogs         = "ServerLogs"       // Copy the server/control and helper logs
+	clientLogs         = "ClientLogs"       // Copy the server/control and helper logs
+	daosConfig         = "ServerConfig"     // Copy the server config
+	customLogs         = "CustomLogs"       // Copy the Custom logs
```

Comment:
If these are all directory names, please be consistent about the variable naming.